mlir/lib/Dialect/Vector/Transforms/VectorLinearize.cpp

```diff
@@ -337,6 +337,9 @@ struct LinearizeVectorExtract final
   matchAndRewrite(vector::ExtractOp extractOp, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
     Type dstTy = getTypeConverter()->convertType(extractOp.getType());
```

Comment:
given `extractOp.getType()` can be a scalar, I wouldn't be surprised if there was a scalar type that could be converted to some other scalar type by this conversion, and thus not return a null type. It seems a bit flaky to assume this type conversion will fail unless it's an n-D vector type, but I can't seem to break it so I'm going to say it's ok, but I just thought I'd mention as this was pretty confusing to me.

Reply:
Yes, me too.

Reply:
I figured it out, the type converter it's using has a single conversion for VectorType at https://github.com/llvm/llvm-project/blob/main/mlir/lib/Dialect/Vector/Transforms/VectorLinearize.cpp#L464-L470

I disregarded that as I couldnt see the connection to `populateVectorLinearizeShuffleLikeOpsPatterns` until I saw https://github.com/llvm/llvm-project/blob/main/mlir/test/lib/Dialect/Vector/TestVectorTransforms.cpp#L867-L868

I guess `convertType` wraps that to return null type if it's unchanged. I think in my head I was thinking of LLVM type converter (which also flattens n-D vectors to 1-D), but of course that's not relevant here.

So yes, it's safe to assume this conversion failing indicates type isn't an n-D vector type :)

Reply:
Okay, I mean.